python/cuml/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -398,160 +596,110 @@ class NearestNeighbors(Base,
         n_jobs=None,  # Ignored, here for sklearn API compatibility
         output_type=None,
     ):
-
-        super().__init__(handle=handle,
-                         verbose=verbose,
-                         output_type=output_type)
-
+        super().__init__(handle=handle, verbose=verbose, output_type=output_type)
         self.n_neighbors = n_neighbors
-        self.n_indices = 0
         self.metric = metric
         self.metric_params = metric_params
         self.algo_params = algo_params
         self.p = p
         self.algorithm = algorithm
-        self._fit_method = self.algorithm
         self.selected_algorithm_ = algorithm
         self.algo_params = algo_params
         self.n_jobs = n_jobs  # Ignored, here for sklearn API compatibility
-        self.knn_index = None
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        # TODO: Indices currently aren't pickleable. For now we drop them and
+        # recreate them on load.
```

Comment:
Previously pickling `NearestNeighbors`/`KNeighborsClassifier`/`KNeighborsRegressor` with `algorithm in {"rbc", "ivfpq", "ivfflat"}` would silently succeed in our tests (where the pointer still existed), but in any realistic use would crash due to invalid pointers on load (this is why you _should not_ store pointers to native objects as integers on python objects).

We now succeed in pickling in these cases, but recreate the indices on load. I don't think this is _wrong_, but it's maybe not ideal. To meet the sklearn conventions we already need to include the original `X` in the serialized state; recreating the index on load takes some time - but serializing the index would increase the binary size since the original data is also there. Something to maybe handle in a followup.

Reply:
I'll create a followup issue. I don't think a warning is necessary in this case - it's not _wrong_ to do it this way, and it's still something we hope to fixup. Warnings IMO are for warning the user about something they can change that will remove the warning (deprecation, parameter out of range moved to threshold, ...). In this case there's not much they can do to adjust the behavior, and it's not giving them incorrect results.